enterprise_knowledge_retriever/src/document_retrieval.py

```diff
@@ -214,7 +218,54 @@ def load_vdb(self, db_path, embeddings):
         vectorstore = self.vectordb.load_vdb(db_path, embeddings, db_type=self.retrieval_info["db_type"])
         return vectorstore
     
-    def get_qa_retrieval_chain(self, vectorstore):
+    def init_retriever(self, vectorstore):
+
+        self.retriever = vectorstore.as_retriever(
+            search_type="similarity_score_threshold",
+            search_kwargs={"score_threshold": self.retrieval_info["score_threshold"], "k": self.retrieval_info["k_retrieved_documents"]},
```

Comment:
Nope.  However, maybe the design can change.

You want to fetch an original high top k, rerank (score and sort), and then select the final k:

https://github.com/snova-codym/ai-starter-kit/blob/007488ee262ce5c9e1a418cf8169a86307af93c8/enterprise_knowledge_retriever/src/document_retrieval.py#L264

Reply:
yes, I mean when you don't want to rerank and select the rerank setting as false, not sure if it will be clear to the user that it will retrieve k_retrieved_documents instead of final_k_retrieved_documents